plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -426,13 +426,19 @@ export class MongoDBInstrumentation extends InstrumentationBase<
     // mongodb is using "tick" when calling a callback, this way the context
     // in final callback (resultHandler) is lost
     const activeContext = context.active();
+    const config = this._config;
     return function patchedEnd(this: {}, ...args: unknown[]) {
       const error = args[0];
       if (error instanceof Error) {
         span.setStatus({
           code: SpanStatusCode.ERROR,
           message: error.message,
         });
+      } else {
+        const results = args[1] as mongodb.CommandCursorResult;
```

Comment:
What if the cast is incorrect? How can we be sure?

Reply:
Adding tests, haven't finished with this PR yet - just pushed before the weekend ;)